typescript/commands/quick_info.py

```diff
@@ -49,17 +56,31 @@ def handle_quick_info(self, quick_info_resp_dict, display_point):
                 status_info_str = info_str + " (^T^Q for more)"
             self.view.set_status("typescript_info", status_info_str)
 
-            # process display string
-            if TOOLTIP_SUPPORT:
-                html_info_str = escape_html(info_str)
-                html_doc_str = escape_html(doc_str)
-                html = "<div>" + html_info_str + "</div>"
-                if len(doc_str) > 0:
-                    html += "<div>" + html_doc_str + "</div>"
-                self.view.show_popup(html, flags=sublime.HIDE_ON_MOUSE_MOVE_AWAY, location=display_point, max_width=800)
         else:
             self.view.erase_status("typescript_info")
 
+        # process tooltips
+        error_html = self.get_error_text_html(display_point)
+        if TOOLTIP_SUPPORT and (info_str != "" or doc_str != "" or error_html != ""):
+            if self.template is None:
+                self.template = Template(load_quickinfo_and_error_popup_template())
+            text_parts = { "error": error_html, "info_str": escape_html(info_str), "doc_str": escape_html(doc_str) }
+            html = self.template.substitute(text_parts)
+            self.view.show_popup(html, flags=sublime.HIDE_ON_MOUSE_MOVE_AWAY, location=display_point, max_height=300, max_width=1500)
+
+    def get_error_text_html(self, pt):
+        client_info = cli.get_or_add_file(self.view.file_name())
+        error_text = ""
+        for (region, text) in client_info.errors['syntacticDiag']:
+            if region.contains(pt):
+                error_text = text
+                break
+        for (region, text) in client_info.errors['semanticDiag']:
```

Comment:
One problem I notice with this approach is that it only handles one error at a time. Also, it requires a full iteration of all the diagnostics. But we could fix this another time.